### PR TITLE
zsh: split documentation into separate outputs

### PR DESCRIPTION
**Pull request by KAction**
###### Description of changes

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/shells/zsh/default.nix [hunk 1]
```diff
@@ -81,6 +82,9 @@ EOF
       ${lib.getBin buildPackages.zsh}/bin/zsh -c "zcompile $out/etc/zprofile"
     ''}
     mv $out/etc/zprofile $out/etc/zprofile_zwc_is_used
+
+    mkdir -p $out/share/doc/
+    mv $out/share/zsh/htmldoc $out/share/doc/zsh-$version
```
Comment (pSub) [inline pkgs/shells/zsh/default.nix:87]: What is the reason for including the $version in the path?

Comment (KAction) [reply pkgs/shells/zsh/default.nix:87]: This way you can build nice `buildEnv` with documentation of different versions.

Comment (pSub) [reply pkgs/shells/zsh/default.nix:87]: Ah. Makes sense!

**Review by pSub — APPROVED** [review]

**Comment by Artturin** [comment]
please check the darwin ci failure

**Comment by pSub** [comment]
@Artturin Thanks for pointing this out. I totally missed those checks. Do you know why they are in status `neutral`. If a command exits with error code 1 I would consider this an error.

**Comment by SuperSandro2000** [comment]
> @Artturin Thanks for pointing this out. I totally missed those checks. Do you know why they are in status `neutral`. If a command exits with error code 1 I would consider this an error.

I don't know either

File: pkgs/shells/zsh/default.nix (hunk 1 above)
Comment (SuperSandro2000) [reply pkgs/shells/zsh/default.nix:87]: Thats a very specific usecase which is normally not required and other files would likely still collide. Please just keep the doc where the build system put it.

Comment (KAction) [reply pkgs/shells/zsh/default.nix:87]: Upstream build system puts them into completely wrong location, $prefix/share/zsh/htmlref.

**Comment by KAction** [comment]
That should be better. Previously documentation was only installed only on linux, now I install it unconditionally.

Built result on my physical darwin, ofBorg seems to be stuck (or overloaded).

**Comment by SuperSandro2000** [comment]
@ofborg eval

**Comment by SuperSandro2000** [comment]
> That should be better. Previously documentation was only installed only on linux, now I install it unconditionally.
> 
> Built result on my physical darwin, ofBorg seems to be stuck (or overloaded).

If ofborg is stuck like this it normally means the API request got somehow lost or was incomplete and we need to trigger it manually.

**Review by SuperSandro2000 — APPROVED** [review]

**Review by pSub — APPROVED** [review]

**Pull request merged**